### PR TITLE
[Windows] Networking fixes (IPv6 Routes Support + No wpcap.dll)

### PR DESCRIPTION
**Pull request by gpotter2**
So as today, IPv6 routes are not supported by scapy on Windows. I guess we can modify that...

This patch:

- Fixes IPv4 get_routes_post2008 function
- Modified version detection to support Windows 10+
- Add the LOOPBACK_NAME on Windows
- Fixes pcapdnet.get_if_raw_hwaddr (on Windows)
- Fixes some route6.py functions to work on Windows
- Removes double warning logs when loading scapy (when winpcap is not installed)

It's not finished because i do not have fully understood what "src candidates" needed to be...

Here is what i can get for now:
http://pastebin.com/xbmR4EHT

**EDIT:**
I finally managed to get it right:
http://pastebin.com/hUU6ykcc

### CODE REVIEW

File: scapy/arch/windows/__init__.py [hunk 1]
```diff
@@ -434,8 +448,72 @@ def read_routes_post2008():
...
+            nh = match.group(4)
+            
+            cset = [] # candidate set (possible source addresses)
+            if iface.name == LOOPBACK_NAME:
+                if d == '::':
+                    continue
+                cset = ['::1']
+            else:
+                devaddrs = filter(lambda x: x[2] == iface, lifaddr)
+                cset = scapy.utils6.construct_source_candidate_set(d, dp, devaddrs, LOOPBACK_NAME)
```
Comment (gpotter2) [inline scapy/arch/windows/__init__.py]: I need some help Here, to understand how to fill `cset`

File: scapy/arch/windows/__init__.py [hunk 2]
```diff
@@ -434,8 +448,72 @@ def read_routes_post2008():
...
+############
+### IPv6 ###
+############
+
+def in6_getifaddr():
+    """
+    Returns all IPv6 addresses found on the computer
+    """
+    ret = []
+    ps = sp.Popen([conf.prog.powershell, 'Get-NetRoute', '-AddressFamily IPV6', '|', 'select ifIndex, DestinationPrefix'], stdout = sp.PIPE, universal_newlines = True)
```
Comment (gpotter2) [inline scapy/arch/windows/__init__.py]: Sample of the command (without the select)
http://pastebin.com/fZPEfzSt

**Comment by guedou** [comment]
Thanks for this effort ! I am really happy to see some activity related to IPv6 on Windows !

To make you life easier and ensure that no regression will ever be introduced, I striongly suggest that you add mocked regression tests as [already done with read_routes6()](https://github.com/secdev/scapy/blob/master/test/regression.uts#L4562) on *BSD. You pastebin example is a good candidate.

**Comment by gpotter2** [comment]
I'm not sure if you should be able to send IPv6 packets or like not yet... I'm gonna add regession tests.

**Comment by gpotter2** [comment]
As i'm was doing networking fixes,
i've added a support when `wpcap.dll` is missing.

It disables the main functions, (maybe there are some more that need to be disabled, but it can be added later)

File: scapy/arch/windows/__init__.py [hunk 3]
```diff
@@ -273,7 +286,7 @@ def load_from_powershell(self):
             except (KeyError, PcapNameNotFoundError):
                 pass
         
-        if len(self.data) == 0:
+        if (len(self.data) == 0) and conf.use_winpcapy:
```
Comment (gpotter2) [inline scapy/arch/windows/__init__.py]: (so it does not say the same thing twice)

**Comment by gpotter2** [comment]
Last patch: regression.uts. @guedou what do you think ?

**Pull request closed without merging**